File: PHP/Patterns/src/Behavioral/Strategy/ExerciseOne/NotificationStrategys/SlackNotificationStrategy.php
<?php

namespace DesignPatterns\Behavioral\Strategy\ExerciseOne\NotificationStrategys;

use DesignPatterns\Behavioral\Strategy\ExerciseOne\Models\User;
use DesignPatterns\Behavioral\Strategy\ExerciseOne\Interfaces\NotificationInterface;


class SlackNotificationStrategy implements NotificationInterface
{
    /**
     * @param User $user
     * @param String $message
     * @return void
     */
    public function sendNotification(User $user, String $message): void
    {
        echo "Sending slack to {$user->getName()} with message: {$message}";
    }
}

File: dart_lang/lib/b_behavioral/b_strategy/c_exercise/notification_strategy/strategies/slack_notification_strategy.dart
import 'package:dart_lang/b_behavioral/b_strategy/c_exercise/notification_strategy/notification_strategy_inteface.dart';
import 'package:dart_lang/b_behavioral/b_strategy/c_exercise/user.dart';

class SlackNotificationStrategy implements NotificationStrategy {
  @override
  void sendNotification(User user, String message) {
    print("sending on Slack... ");
  }
}
